From bf9f959ec23b94fa24509a09fe33e3ed0c341720 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Tue, 01 Nov 2022 13:02:00 +0200
Subject: [PATCH] [SW-108910] gaudi2: set unique revision for bis compliant hl225

For identfication of new SKU, the PCIE DBI revision id
will be changed to 0x2

Change-Id: Ic4ccb108712c1806327f9288a572537290c1165f
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index c4430d5..30213d7 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -511,9 +511,12 @@
 	preboot_init_pcie_phy_fw_update();
 }
 
+#define HL_PCIE_REVISION_ID_MASK	0xFF
+#define HL225B_PCIE_REVISION_ID		0x2
 static void preboot_pcie_device_id_vendor_id_set(void)
 {
 	u32 device_id_vendor_id = hl_pci_get_ven_id_dev_id();
+	u32 class_rev;
 
 	/* Update Vendor ID */
 	regmap_writel(device_id_vendor_id,
@@ -523,6 +526,17 @@
 	regmap_writel(device_id_vendor_id,
 		      mmPCIE_DBI_SUBSYSTEM_ID_SUBSYSTEM_VENDOR_ID_REG);
 
+	/* Update REV ID for HL225B */
+	if (preboot_check_bis_compliant_mode()) {
+		class_rev = regmap_readl(mmPCIE_DBI_CLASS_CODE_REVISION_ID);
+
+		/* clear low byte in PCIE class low word */
+		class_rev &= ~(HL_PCIE_REVISION_ID_MASK);
+		class_rev |= HL225B_PCIE_REVISION_ID;
+
+		regmap_writel(class_rev, mmPCIE_DBI_CLASS_CODE_REVISION_ID);
+	}
+
 }
 
 void preboot_init_pcie(void)
diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index d48dbfb..c0d13bd 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -1498,3 +1498,20 @@
 	/* reset ARC */
 	reset_arc_farm_arc0();
 }
+
+bool preboot_check_bis_compliant_mode(void)
+{
+	const struct hl_sec_efuse *efuse;
+
+	if (preboot_fake_security_is_enabled())
+		efuse = preboot_get_hl_sec_fake_efuse();
+	else
+		efuse = preboot_get_hl_sec_real_efuse();
+
+	if (!efuse) {
+		preboot_puts("failed to read efuse, forcing bis compliant mode.\n");
+		return true;
+	}
+
+	return efuse->bis_compliant;
+}
diff --git a/include/preboot.h b/include/preboot.h
index 3da58cf..9503f22 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -182,6 +182,7 @@
 void preboot_dynamic_img_elbi_reg_cleanup(void *map_addr, u32 size);
 bool preboot_is_overlap_address_range(u64 addr, int len);
 
+bool preboot_check_bis_compliant_mode(void);
 enum hl_fw_en_caps;
 
 void preboot_dev_enable_caps(void);
